### PR TITLE
719 Fix: Clear list error when adding a list item to a list

### PR DESCRIPTION
**Pull request by adamdavies1**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
Please also include any acceptance criteria if you have any.

- Fix Error message on list creation
- [Link to the related issue and recreation steps](https://github.com/XGovFormBuilder/digital-form-builder/issues/719)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce
the testing if necessary.

Before PR's can be merged they will need to be tested by QA and approved where
applicable. To flag the change to QA assign **@XGovFormBuilder/qa** as one of the reviewers.

- To Reproduce
1. Go to the designer
2. lists
3. create a list with no items
4. click save
5. see error message
6. create a list item and save
7. error message should no longer persist

# Checklist:

- [x] My changes do not introduce any new linting errors
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto main and there are no code conflicts
- [x] I have checked deployments are working in all environments

### CODE REVIEW

File: smoke-tests/designer/wdio.conf.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const { hooks } = require("./support/hooks");
 const drivers = {
-  chrome: { version: "98.0.4758.102" }, // https://chromedriver.chromium.org/
+  chrome: { version: "100.0.4896.60" }, // https://chromedriver.chromium.org/
```
Comment (RichardBray) [inline smoke-tests/designer/wdio.conf.js:3]: I've seen this change in [another PR](https://github.com/XGovFormBuilder/digital-form-builder/pull/829/files#diff-4335e7563ea8c424c5534b178892f25967932dcb68161560998a75ef993863c7R3)

Should there be a separate PR just to fix this specific issue?

File: designer/client/list/ListEdit.tsx [hunk 2]
```diff
@@ -98,12 +98,20 @@ function useListEdit() {
   };
 }
 
+function validate(errors: any, selectedList: any) {
```
Comment (RichardBray) [inline designer/client/list/ListEdit.tsx:101]: Are we okay with anys in this project?

It's fine if we are I'm just checking.

File: smoke-tests/designer/wdio.conf.js (hunk 1 above)
Comment (jenbutongit) [reply smoke-tests/designer/wdio.conf.js:3]: actually, yes, that would be useful ☝️ (it's to do with wdio/selenium.. I'm sure you've heard enough of my rants about it so I won't bore you again lol)

Comment (adamdavies1) [reply smoke-tests/designer/wdio.conf.js:3]: Yeah iv been meaning to bring this up, is there no way of auto checking the latest version to avoid us all having to manually update? (question for the issue perhaps)

File: designer/client/list/ListEdit.tsx (hunk 2 above)
Comment (adamdavies1) [reply designer/client/list/ListEdit.tsx:101]: Yes i think so, it seems to be used throughout

Comment (RichardBray) [reply designer/client/list/ListEdit.tsx:101]: Okay in that case it's fine but going forward I think it would make sense to try to avoid using `any` and use the proper types. It might be more work but it's definitely worth it in the long run.

**Review by RichardBray — APPROVED** [review]
